src/config.rs

```diff
@@ -166,6 +166,12 @@ impl IndexMut<usize> for ColorList {
     }
 }
 
+#[derive(Debug, Deserialize)]
+pub struct Shell {
+    program: Option<String>,
```

Comment:
Let's enforce that if `shell` is specified in the config, that `program` *must* be `Some`.

Holding a `Cow<'a, str>` instead of a `String` would give us a little more flexibility with the type (see comment in tty.rs).

We'll probably want a helper method on this like

```rust
fn program(&self) -> &str;
```

So consumers don't need to deal with the `Cow`.